Safety/Src/interchip_S.cpp

```diff
@@ -1,93 +1,57 @@
-#if 0
+#include "spi.h"
+#include "Interchip_S.hpp"
 
-/*
-#include "interchip_S.h"
-//#include "debug.h"
-#include <stdio.h>
-#include "spi.c"
+static volatile Interchip_Packet rxData;
+static volatile Interchip_Packet txData;
+static volatile bool dataNew;
 
-
-static Interchip_StoA_Packet* dataTX;
-static Interchip_AtoS_Packet* dataRX;
-static uint16_t errorCount=0;
-//
-
-uint16_t getErrorCount(){
-  return errorCount;
+// Get the pwm signal that has been recieved.
+int16_t getPWM(int8_t index) {
```

Comment:
Can you actually just return the whole struct 1 shot ? Makes things a bit easier, also easier in terms of the isDataNew stuff, which this function may as well return.

Reply:
oh yeah I hadnt considered the isDataNew stuff, will do. Should I be returning the entire struct or just the PWM array?

Reply:
it now returns the array